### PR TITLE
docs: add release process document

### PR DESCRIPTION
**Pull request by phlogistonjohn**
Finally, I've gotten around to documenting the release process based on my notes from the release in June. This is loosely based on the release process doc I wrote for sambacc and svcwatch.

### CODE REVIEW

File: docs/release-process.md
Comment (obnoxxx) [inline docs/release-process.md:107]:
```suggestion
Images built for this release can be obtained  from the quay.io image registry.
```

File: docs/release-process.md
Comment (obnoxxx) [inline docs/release-process.md:30]: Does this mean to push changes directly instead of using pull requests? Ig yes, I think it would be helpful to mention this explicitly ...

**Review by obnoxxx — CHANGES_REQUESTED** [review]
Great doc, thanks @phlogistonjohn  !

Just two minor requests.

File: docs/release-process.md
Comment (phlogistonjohn) [reply docs/release-process.md]: The words are pretty much synonyms in my head, but sure. Done.

File: docs/release-process.md
Comment (phlogistonjohn) [reply docs/release-process.md:30]: Added & expanded upon.

File: docs/release-process.md
Comment (obnoxxx) [reply docs/release-process.md]: > The words are pretty much synonyms in my head, but sure. Done.

Thanks! I understood the intention originally but couldn't resist because "acquire" always sounds to me like costs are involved ;-)

**Review by obnoxxx — APPROVED** [review]
Thanks for addressing the requests!LGTM

**Comment by obnoxxx** [comment]
@mergifyio refresh

**Comment by obnoxxx** [comment]
@phlogistonjohn  apologies: I merged without waiting for a second approval. afraid I can't fix that any more now ...

**Comment by phlogistonjohn** [comment]
OK, others can file issues or PRs as needed.

**Pull request merged**
